src/btree/bt_sync.c

```diff
@@ -80,6 +81,19 @@ __sync_file(WT_SESSION_IMPL *session, int syncop)
 		break;
 	case WT_SYNC_CHECKPOINT:
 		/*
+		 * If we are flushing a file at read-committed isolation, which
+		 * is of particular interest for flushing the metadata to make
+		 * schema-changing operation durable, get a transactional
+		 * snapshot now.
+		 *
+		 * All changes committed up to this point should be included.
+		 * We don't update the snapshot in between pages because (a)
+		 * the metadata shouldn't be that big, and (b) if we do ever
```

Comment:
I noticed that this comment is incomplete, but on checking it is also incomplete on develop.  I'll open a ticket for that...